zookeeper-server/src/main/java/org/apache/zookeeper/server/DataTree.java

```diff
@@ -366,6 +376,30 @@ public void addConfigNode() {
         }
     }
 
+    /**
+     * create a /zookeeper/uri-domain-map node for maintaining client-URIs to domain mapping for
+     * zookeeper
+     */
+    public void addUriDomainMapNode() {
+        if (!X509AuthenticationConfig.getInstance().isX509ZnodeGroupAclEnabled()) {
+            return;
+        }
+        DataNode zookeeperZnode = nodes.get(procZookeeper);
+        if (zookeeperZnode != null) { // should always be the case
+            zookeeperZnode.addChild(uriDomainMapChildZookeeper);
+        } else {
+            assert false : "There's no /zookeeper znode - this should never happen.";
+        }
```

Comment:
A few comments:
1. In Java, we usually throw exceptions over `assert`s.
2. I'm not sure if that check is necessary here - like you said, `zookeeperZNode` should always exist? Then can we simplify things?

Reply:
+1, if /zookeeper doesn't exist, we should let it fail.

Reply:
In general i agree with your thoughts but here i was just trying to follow things just like addConfigNode method. I don't have full context on these checks but i'm fine with keeping it just like addConfigNode method or changing it completely.